File: src/main/Leetcode00003.go
package main

func lengthOfLongestSubstring(s string) int {
	if len(s) == 0 {
		return 0
	}
	i, j := 0, 0
	m := 0
	w := make(map[byte]bool, len(s))
	for i < len(s) && j < len(s) {
		c := s[j]
		if _, ok := w[c]; !ok {
			w[c] = true
			if len(w) > m {
				m = len(w)
			}
			j++
		} else {
			delete(w, s[i])
			i++
		}
	}
	return m
}

File: src/main/Leetcode00003.java
import java.util.*;

public class Solution {
    public int lengthOfLongestSubstring(String s) {
        Integer mapIndexByCharacter[] = new Integer[128];
        int max = 0, j = 0;
        for (int i = 0;i < s.length();i++) {
            Integer index = mapIndexByCharacter[s.charAt(i)];
            if (index != null) {
                j = Math.max(j, index+1);
            } 
            mapIndexByCharacter[s.charAt(i)] = i;
            max = Math.max(max, i - j + 1);
        }
        return max;
    }
}
